### PR TITLE
Add cutoff option to amp-timeago

### LINKED ISSUE #9415: ITT: amp-timeago add cutoff time
We have a use case where we want the date to be formatted as time ago only for a defined time period e.g. if the time ago is over a week then display the date rather than X days ago.

This ITT is to add support for a cutoff property of the timeago component.

e.g.
<amp-timeago layout="fixed-height" height="20" datetime="2017-04-11T00:37:33.809Z" locale="en" **cutoff="604800000"**>Saturday 11 April 2017 00.37</amp-timeago>

So if the datetime was over the cutoff ago 604800000 milliseconds (1 week), it would not convert to timeago. 

@edhollinghurst FYI

### PR DESCRIPTION
**Pull request by edhollinghurst**
This pull request add a cutoff option to the amp-timeago component as discussed in #9415.

I've also made a tweak to presubmit-checks.js. Is it OK for amp-timeago to be whitelisted for the getTime() method?

Closes #9415.

### CODE REVIEW

File: extensions/amp-timeago/0.1/test/test-amp-timeago.js [hunk 1]
```diff
@@ -27,19 +27,29 @@ describes.realWin('amp-timeago', {
...
+    const date = new Date();
+    date.setDate(date.getDate() - 2);
+    element.setAttribute('datetime', date.toISOString());
+    element.textContent = date.toString();
     element.build();
     const timeElement = element.querySelector('time');
     expect(timeElement.textContent).to.equal('2 days ago');
   });
+
+  it('should display original date when built', () => {
```
Comment (jridgewell) [inline extensions/amp-timeago/0.1/test/test-amp-timeago.js]: "should display original date when older than cutoff"

File: extensions/amp-timeago/amp-timeago.md [hunk 2]
```diff
@@ -94,6 +94,10 @@ By default, the local is set to <code>en</code>; however, you can specify one of
   <li>zhTW (Taiwanese)</li>
 </ul>
 
+**cutoff** (optional)
+
+Display the original date if time distance is older than cutoff (seconds).
```
Comment (jridgewell) [inline extensions/amp-timeago/amp-timeago.md]: /cc @bpaduch

File: extensions/amp-timeago/validator-amp-timeago.protoascii [hunk 3]
```diff
@@ -36,6 +36,7 @@ tags: {  # <amp-timeago>
     value_regex: "\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d(:[0-5]\\d(\\.\\d+)?)?Z"
   }
   attrs: { name: "locale" }
+  attrs: { name: "cutoff" }
```
Comment (jridgewell) [inline extensions/amp-timeago/validator-amp-timeago.protoascii]: Needs a `value_regex: "\\d+"`. /cc @honeybadgerdontcare

File: extensions/amp-timeago/0.1/amp-timeago.js [hunk 4]
```diff
@@ -43,13 +46,29 @@ export class AmpTimeAgo extends AMP.BaseElement {
...
-    timeElement.textContent = timeago(this.datetime_, this.locale_);
+
+    if (this.cutoff_) {
+      const elDate = new Date(this.datetime_);
+      const secondsAgo = Math.floor((Date.now() - elDate.getTime()) / 1000);
+
+      if (secondsAgo > this.cutoff_) {
+        timeElement.textContent = this.title_;
+      }
+      else {
```
Comment (jridgewell) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Joined with the `}`, please.

File: extensions/amp-timeago/0.1/amp-timeago.js [hunk 5]
```diff
@@ -43,13 +46,29 @@ export class AmpTimeAgo extends AMP.BaseElement {
...
+        timeElement.textContent = timeago(this.datetime_, this.locale_);
+      }
+    }
+    else {
```
Comment (jridgewell) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Joined with the `}`, please.

File: extensions/amp-timeago/0.1/amp-timeago.js [hunk 6]
```diff
@@ -32,6 +32,9 @@ export class AmpTimeAgo extends AMP.BaseElement {
 
     /** @private {string} */
     this.title_ = '';
+
+    /** @private {string} */
```
Comment (jridgewell) [inline extensions/amp-timeago/0.1/amp-timeago.js]: This should be a `{number}`. Let's define to `Infinity` as a default.

File: extensions/amp-timeago/0.1/amp-timeago.js [hunk 7]
```diff
@@ -43,13 +46,29 @@ export class AmpTimeAgo extends AMP.BaseElement {
     this.locale_ = this.element.getAttribute('locale') ||
       this.win.document.documentElement.lang;
     this.title_ = this.element.textContent;
+    this.cutoff_ = parseInt(this.element.getAttribute('cutoff'), 10);
```
Comment (jridgewell) [inline extensions/amp-timeago/0.1/amp-timeago.js]: We don't want to do this unless it really has the `cutoff` attribute.

File: extensions/amp-timeago/validator-amp-timeago.protoascii (hunk 3 above)
Comment (honeybadgerdontcare) [inline extensions/amp-timeago/validator-amp-timeago.protoascii]: Seems reasonable. @Gregable as fyi for pulling in.

File: extensions/amp-timeago/amp-timeago.md (hunk 2 above)
Comment (edhollinghurst) [inline extensions/amp-timeago/amp-timeago.md]: Thanks - that's now been updated.

File: extensions/amp-timeago/validator-amp-timeago.protoascii (hunk 3 above)
Comment (edhollinghurst) [inline extensions/amp-timeago/validator-amp-timeago.protoascii]: Thanks  - I've now added in `value_regex: "\\d+"`.

File: extensions/amp-timeago/0.1/amp-timeago.js (hunk 6 above)
Comment (edhollinghurst) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Thanks - I've now made this type `{?number}` with `null` as the default.

File: extensions/amp-timeago/0.1/amp-timeago.js (hunk 7 above)
Comment (edhollinghurst) [inline extensions/amp-timeago/0.1/amp-timeago.js]: OK I've made a change so this is only done if the attribute exists.

File: extensions/amp-timeago/0.1/amp-timeago.js (hunk 4 above)
Comment (edhollinghurst) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Done

File: extensions/amp-timeago/0.1/amp-timeago.js (hunk 5 above)
Comment (edhollinghurst) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Done

File: extensions/amp-timeago/validator-amp-timeago.protoascii [hunk 8]
```diff
@@ -36,6 +36,10 @@ tags: {  # <amp-timeago>
     value_regex: "\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d(:[0-5]\\d(\\.\\d+)?)?Z"
   }
   attrs: { name: "locale" }
+  attrs: {
```
Comment (honeybadgerdontcare) [inline extensions/amp-timeago/validator-amp-timeago.protoascii]: Could you move this above "datetime"? We try to keep the attributes in alphabetical order. Thanks!

Comment (edhollinghurst) [inline extensions/amp-timeago/validator-amp-timeago.protoascii]: Sure - that's done.

File: extensions/amp-timeago/0.1/amp-timeago.js [hunk 9]
```diff
@@ -49,7 +52,21 @@ export class AmpTimeAgo extends AMP.BaseElement {
 
     const timeElement = document.createElement('time');
     timeElement.setAttribute('datetime', this.datetime_);
-    timeElement.textContent = timeago(this.datetime_, this.locale_);
+
+    if (this.element.hasAttribute('cutoff')) {
+      this.cutoff_ = parseInt(this.element.getAttribute('cutoff'), 10);
```
Comment (jridgewell) [inline extensions/amp-timeago/0.1/amp-timeago.js]: The class member shouldn't be necessary now.

Comment (edhollinghurst) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Should I change `this.cutoff_` to be a variable?

Comment (jridgewell) [inline extensions/amp-timeago/0.1/amp-timeago.js]: Yup.

Comment (edhollinghurst) [inline extensions/amp-timeago/0.1/amp-timeago.js]: OK that's done.

**Comment by aghassemi** [comment]
@jridgewell good to merge?

**Pull request merged**